File: src/js/test/version.test.js
import { describe, expect, test } from 'vitest';
import { formatVersion } from '@/utils/version.js';

describe('formatVersion', () => {
  const build = {
    commit: 'abc1234',
    buildDate: '2026-09-09 01:43:21 UTC',
  };

  test('includes a release version with commit and build date', () => {
    expect(formatVersion({ ...build, tag: 'v2026.08.26' })).toBe(
      'Version v2026.08.26 (commit abc1234, built on 2026-09-09 01:43:21 UTC)',
    );
  });

  test('includes an upstream branch name with commit and build date', () => {
    expect(
      formatVersion({
        ...build,
        tag: 'main',
        repo: 'sandialabs/sceptre-phenix',
      }),
    ).toBe('Branch main (commit abc1234, built on 2026-09-09 01:43:21 UTC)');
  });

  test('includes a fork repo with branch builds', () => {
    expect(
      formatVersion({
        ...build,
        tag: 'main',
        repo: 'GhostofGoes/sceptre-phenix',
      }),
    ).toBe(
      'Branch main [ghostofgoes/sceptre-phenix] (commit abc1234, built on 2026-09-09 01:43:21 UTC)',
    );
  });
});

File: src/go/version/version_test.go
package version

import "testing"

func TestLabel(t *testing.T) {
	originalRepo := Repo
	originalTag := Tag
	defer func() {
		Repo = originalRepo
		Tag = originalTag
	}()

	tests := []struct {
		name string
		repo string
		tag  string
		want string
	}{
		{name: "release", repo: upstreamRepo, tag: "v2026.08.26", want: "Version v2026.08.26"},
		{name: "upstream branch", repo: upstreamRepo, tag: "main", want: "Branch main"},
		{name: "fork branch", repo: "GhostofGoes/sceptre-phenix", tag: "main", want: "Branch main [ghostofgoes/sceptre-phenix]"},
	}

	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			Repo = test.repo
			Tag = test.tag
			if got := Label(); got != test.want {
				t.Fatalf("Label() = %q, want %q", got, test.want)
			}
		})
	}
}
